[IMP] website: adapt alerts colors and options

qsm-odoo opened the pull request:
Extracted from https://github.com/odoo/odoo/pull/120302

Following the default theme redesign, this commit adapts the color of the alert component and reviews its options.

task-3888927

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/137956).

qsm-odoo commented:
@stefanorigano Reviving this :) I rebased the branch. Before going further, is this still what we want? (maybe specs changed during snippets refactoring?). If yes, then I'll assign a dev to finish this :+1:

anso-odoo commented:
Hello @qsm-odoo. I had a look and this is still a much appreciated and needed IMP. 

While checking I found a small issue with borders, which I fixed (basically a "border" property in `s_alert.scss` with no set border-color was making it inherit the currentColor and thus making it too dark). I also tweaked a bit the background behind the icon to make it a bit more sublte. Added these changes in a fixup so you can have a look. 

Thanks !

cc @stefanorigano 

Before: 
<img width="665" alt="Screenshot 2024-04-18 at 11 29 35" src="https://github.com/odoo/odoo/assets/110090660/2ffe2d35-1a93-40ad-a288-f1dbe300a21d">

After: 
<img width="653" alt="Screenshot 2024-04-18 at 11 25 07" src="https://github.com/odoo/odoo/assets/110090660/2811d577-6571-4887-b5b7-5fa7986d11c7">

qsm-odoo commented:
Rebased (fixed conflicts), squashed commits. I will now ask a dev to look into this (to make sure old alerts are treated the way it should, the code is ok, etc).

xO-Tx commented on addons/website/static/src/snippets/s_alert/options.js:
```diff
@@ -0,0 +1,30 @@
+        // targeting the main container. This is also why this does not need
+        // a _computeWidgetState, relying on the "selectClass" one which comes
+        // afterwards alphabetically (compared to "selectAlertIcon").
+        alertIconEl.classList.remove(...params.possibleValues.filter(v => !!v));
```
@bvr-odoo It seems that this class removal doesn't handle all possible scenarios... If we remove only the `possibleValues` and we set a different icon from the media dialog, we will end up with an element holding more than one `fa-` class, which leads to inconsistencies :grimacing:

https://github.com/odoo/odoo/assets/25032876/750baed1-b7a6-4f53-8a60-185e6acea6a4

We simply need to remove all icons classes before setting the right one :+1:

qsm-odoo commented:
@bvr-odoo There is no alert in the codebase where you need to increase the version number? Instead of increasing the version number, not possible to make a migration script? To confirm :)

@robodoo delegate=SergeBayet 
@SergeBayet

bvr-odoo commented:
> @bvr-odoo There is no alert in the codebase where you need to increase the version number?

No

> Instead of increasing the version number, not possible to make a migration script? To confirm :)

I'm not sure about to change the design of existing pages.
If they wish, users can change their “alert” themselves. But I wouldn't change them automatically
IMO this seems a bit "intrusive"
But I could be wrong :)

SergeBayet reviewed (commented):
Good PR :+1: 

As a reminder, the task-id should be updated in both the PR description and in the commit message. :smiley: 

LGTM but I have two questions regarding the UI:

1. When a link is inserted into an alert snippet, it lacks the "alert-link" class, which ensures the link color matches the alert theme. Is this intentional, or can a CSS rule be added to apply this class, such as:
   ```css
   a {
     font-weight: 600;
     color: var(--alert-link-color);
   }
   ```

2. Shouldn't the icon be vertically aligned as in the Bootstrap documentation? 
![image](https://github.com/odoo/odoo/assets/22252071/732534f6-a632-4863-a880-b2fc98cb09e3)

bvr-odoo commented:
> 1- When a link is inserted into an alert snippet, it lacks the "alert-link" class, which ensures the link color matches the alert theme. Is this intentional, or can a CSS rule be added to apply this class, such as:

I won't do that, it will impact inner snippet dropped inside the "alert"

> 2- Shouldn't the icon be vertically aligned as in the Bootstrap documentation?

I can't reproduce it.. it's not the style of the new alert in your screenshot

SergeBayet commented on addons/website/static/src/snippets/s_alert/options.js:
```diff
@@ -0,0 +1,40 @@
+     * @see this.selectClass for parameters
+     */
+    selectAlertIcon(previewMode, widgetValue, params) {
+        const alertIconEl = this.$target[0].querySelector('.s_alert_icon');
```
double quotes here :)

SergeBayet commented on addons/website/static/src/snippets/s_alert/options.js:
```diff
@@ -0,0 +1,40 @@
+/** @odoo-module **/
```
Not mandatory anymore

SergeBayet commented on addons/website/static/src/snippets/s_alert/options.js:
```diff
@@ -0,0 +1,40 @@
+/** @odoo-module **/
+
+import options from "@web_editor/js/editor/snippets.options";
+import fonts from '@web_editor/js/wysiwyg/fonts';
```
double quotes here.

SergeBayet commented on addons/website/static/src/snippets/s_alert/options.js:
```diff
@@ -0,0 +1,40 @@
+import fonts from '@web_editor/js/wysiwyg/fonts';
+
+options.registry.Alert = options.Class.extend({
+
```
My linter tells me to remove this blank line...

bvr-odoo replied:
don't trust him 😸
There is always empty lignes before and after this type of comment section.
![image](https://github.com/user-attachments/assets/9c16637d-1d3f-4adb-a240-e10c553699fe)

SergeBayet commented on addons/website/static/src/snippets/s_alert/options.js:
```diff
@@ -0,0 +1,40 @@
+        if (previewMode === "reset") {
+            iconClass = this.initialIconClass;
+        } else {
+            this.initialIconClass = [...alertIconEl.classList]
+                .filter(c => allFaIcons.includes(c));
```
```suggestion
            this.initialIconClass = [...alertIconEl.classList].filter((c) =>
                allFaIcons.includes(c)
            );
```

bvr-odoo replied:
👍🏼 I changed it a bit differently than you suggested

SergeBayet reviewed (commented):
LTGTM :+1: Just some lintering remarks.

SergeBayet reviewed (approved):
Thanks for the modifications. LGTM :)

SergeBayet commented:
@robodoo r+
